src/ImageSharpCommunity.Providers.Remote/RemoteImageProvider.cs

```diff
@@ -60,4 +64,30 @@ private bool IsMatch(HttpContext context)
     {
         return context.Request.Path.GetMatchingRemoteImageProviderSetting(_options) != null;
     }
+
+    private bool UrlReturnsSuccess(RemoteImageProviderSetting setting, Uri uri)
+    {
+        if (setting.VerifyUrl == false)
+        {
+            _logger.LogDebug("Skipping verification of URL {Url} as VerifyUrl is set to false", uri);
+            return true;
+        }
+
+        if (_cache.TryGetValue(nameof(RemoteImageProvider) + uri, out bool cachedResult))
+        {
+            _logger.LogDebug("Using cached result for URL {Url}", uri);
+            return cachedResult;
+        }
+
+        var client = _clientFactory.GetRemoteImageProviderHttpClient(setting);
+        var request = new HttpRequestMessage(HttpMethod.Head, uri);
+        var response = client.SendAsync(request).Result;
+
+        if (response.Headers.CacheControl?.MaxAge is not null)
+        {
+            _cache.Set(nameof(RemoteImageProvider) + uri, response.IsSuccessStatusCode, response.Headers.CacheControl.MaxAge ?? TimeSpan.Zero);
```

Comment:
I wonder if the fallback value should be higher than TimeSpan.Zero, we would expect the header to always exist, but if it doesn't for some reason then setting the fallback cache expiration should maybe be an hour or something just to avoid more spamming?

Reply:
There is a setting for disabling it (https://github.com/skttl/ImageSharpCommunity.Providers.Remote/pull/4/files/eebcbca5a3b5cf3200aa732c6731e48e47d375ee#diff-12880d2b1b3960431556d39eb33890821d623cb973bfaf324a888b31e69206f0R63) its enabled by default :)